src/ui/components/filters/filteroptionscomponent.js

```diff
@@ -295,31 +320,18 @@ export default class FilterOptionsComponent extends Component {
      * @type {boolean}
      */
     this.showMoreState = this.config.showMore;
+  }
 
-    if (this.config.storeOnChange) {
-      this.apply();
-    }
-
-    if (!this.config.isDynamic) {
-      this.core.storage.registerListener({
-        eventType: 'update',
-        storageKey: StorageKeys.HISTORY_POP_STATE,
-        callback: historyData => {
-          const data = historyData.get(this.name);
-          try {
-            const newOptions = data ? JSON.parse(data) : data;
-            this.config.options = this.config.getSelectedOptions(
-              this.config.initialOptions,
-              newOptions
-            );
-            this.updateListeners(false, true);
-            this.setState();
-          } catch (e) {
-            console.warn(`Filter option ${data} could not be parsed:`, e);
-          }
-        }
-      });
-    }
+  _registerBackNavigationListener () {
+    this.core.storage.registerListener({
+      eventType: 'update',
+      storageKey: StorageKeys.HISTORY_POP_STATE,
+      callback: () => {
+        this._initVariables(this._config);
+        this.updateListeners(true, true);
```

Comment:
the first param, `alwaysSaveFilterNodes`, was changed from false to true so that back navigation would work for searchOnChange = false FilterBoxes. When searchOnChange = true the behavior is the same